### PR TITLE
Fix build

### PR DESCRIPTION
**Pull request by KarishmaBothara**
I have looked at the polkadots current UI repo at https://github.com/polkadot-js/apps to fix most of the typescript errors..
Have removed the packages we don't support yet for example staking, council, treasury.
There are two places where I ignore typescript error, one of they are staking related..which I will check while working on staking.. The other place where I have added it is for a styled component.. I am looking to fix this one.. 
But the diff is huge.. so thought to create PR while I fix that one.

### CODE REVIEW

**Review by holygits — APPROVED** [review]

**Comment by KarishmaBothara** [comment]
Need to fix lint.. investigating the issues with yarn lint.. trying to make circle ci build happy..

File: packages/react-api/src/ledger.ts [hunk 1]
```diff
@@ -31,8 +31,14 @@ export function clearLedger (): void {
 
 export function getLedger (): Ledger {
   if (!ledger) {
-    ledger = new Ledger(uiSettings.ledgerConn as 'u2f');
+    const def = api && ALLOWED_CHAINS.find(([g]) => g === api.genesisHash.toHex());
```
Comment (alexsednz) [inline packages/react-api/src/ledger.ts:34]: Would allowed_chains now restrict us to kusama?

Comment (KarishmaBothara) [reply packages/react-api/src/ledger.ts:34]: no.. I will update allowed chains to include polkadot too, as 
const ALLOWED_CHAINS: [string, 'kusama' | 'polkadot'][] = [
  [KUSAMA_GENESIS, 'kusama'],
  [POLKADOT_GENESIS, 'polkadot']
];

File: packages/apps/src/SideBar/SideBarScroll.tsx [hunk 2]
```diff
@@ -7,6 +7,7 @@ const SideBarScroll = styled.div.attrs({
   @media (min-width:1000px){
     .expanded & {
       position: fixed;
+      width: 12%;
```
Comment (KarishmaBothara) [inline packages/apps/src/SideBar/SideBarScroll.tsx:10]: Updated the css property to fix the issue

**Comment by KarishmaBothara** [comment]
Fixed all the issues mentioned [here](https://centralitydev.atlassian.net/browse/IND-360?focusedCommentId=23339) as well fixed the lint issue by adding resolution for the following packages.. 
     "@polkadot/dev": "0.40.19",
     "@babel/cli": "7.8.4",
     "@babel/code-frame": "7.8.3",
     "@babel/compat-data": "7.8.4",
     "@babel/core": "7.8.4",
     "@babel/generator": "7.8.4",

@polkadot/keyring internally installs @polkadot/dev at 0.40.22 and updated babel packages which was causing the lint issues..

File: packages/react-api/src/ledger.ts [hunk 3]
```diff
@@ -5,17 +5,32 @@
...
+}
 
-const ALLOWED_CHAINS = chains.kusama;
+const KUSAMA_GENESIS = getGenesis('kusama');
+
+const POLKADOT_GENESIS = getGenesis('polkadot');
+
+const ALLOWED_CHAINS: [string, 'kusama' | 'polkadot'][] = [
+  [KUSAMA_GENESIS, 'kusama'],
+  [POLKADOT_GENESIS, 'polkadot']
```
Comment (alexsednz) [inline packages/react-api/src/ledger.ts:26]: What about Azalea and Nikau?

**Comment by alexsednz** [comment]
Looks a nice work to me as a non-expert!

**Review by alexsednz — APPROVED** [review]

File: packages/react-api/src/ledger.ts (hunk 3 above)
Comment (KarishmaBothara) [reply packages/react-api/src/ledger.ts:26]: ya.. I am looking at https://github.com/polkadot-js/common/blob/master/packages/networks/src/index.ts package to see how they use the KUSAMA and POLKADOT networks genesis hash.. I am also trying to understand when this Ledger is been used..

Comment (KarishmaBothara) [reply packages/react-api/src/ledger.ts:26]: Right now as we don't support hardware type...we won't add support for Azalea n Nikau at this stage.

**Comment by holygits** [comment]
Working well from my tests so far.
One minor thing looks odd on the sidebar (firefox)
Screen Shot 2020-11-03 at 13 20 25

File: packages/app-explorer/src/Events.tsx [hunk 4]
```diff
@@ -56,7 +56,7 @@ function Events ({ emptyLabel, eventClassName, events, withoutIndex, t }: Props)
               <summary>
                 {
                   event.meta && event.meta.documentation
-                    ? event.meta.documentation.join(' ')
+                    ? event.meta.documentation.join(' ').split('\\[info\\]').join(' ')
```
Comment (AlexZhenWang) [inline packages/app-explorer/src/Events.tsx:59]: I guess we can also use `replace(//g, '' )` to replace all, but yeah, in this case we will need to write code like `a.replace(/\\\[info\\\]/g,"rr")` to process the special character `[` and `]`, which make `a.replace(/\\\[info\\\]/g,"rr")` looks confuse T T. love your implement 👍

**Review by AlexZhenWang — APPROVED** [review]
LGTM~

**Comment by holygits** [comment]
I'm in favour of merging this, small stylistic fixes can be done in future PRs

**Pull request merged**